WebDriver. Task 3.

mccreed13 opened the pull request.

mccreed13 committed:
Added pages, test and enums. Code is not work.

vaddost commented on src/main/java/com/epam/training/student_anton_lapushenko/WebDriver/Task3/page/GoogleCloudCalculatorPage.java:
```diff
@@ -0,0 +1,127 @@
+import java.time.Duration;
+import java.util.List;
+
+public class GoogleCloudCalculatorPage {
```
Move all properties to the top of the page before the methods

vaddost commented on src/main/java/com/epam/training/student_anton_lapushenko/WebDriver/Task3/page/GoogleCloudCalculatorPage.java:
```diff
@@ -0,0 +1,127 @@
+        computerEngineButton.click();
+    }
+
+    private final By countInstancesBy = By.xpath("(//*[@class='qdOxv-fmcmS-yrriRe qdOxv-fmcmS-yrriRe-OWXEXe-INsAgc qdOxv-fmcmS-yrriRe-OWXEXe-di8rgd-V67aGc qdOxv-fmcmS-yrriRe-OWXEXe-V67aGc-NLUYnc'])[1]");
```
try to avoid using of generated class names in your locators. These names can be dynamic and could change that would produce a lot of auto issues in future

vaddost commented on src/main/java/com/epam/training/student_anton_lapushenko/WebDriver/Task3/page/GoogleCloudSearchPage.java:
```diff
@@ -0,0 +1,25 @@
+public class GoogleCloudSearchPage {
+    WebDriver driver;
+
+    private final By searchCalculatorResultBy = By.xpath("/html/body/c-wiz[2]/div/div/div/div/div/div[3]/c-wiz/div[1]/div[1]/div/div");
```
try to avoid using the absolute path in xpath locators

vaddost replied:
in this case the locator //a[@track-type='search-result'] could work. You can find all this elements and click on first result from list
Also, you need to add wait before calling findElements() method

vaddost commented on src/test/java/com/epam/training/student_anton_lapushenko/WebDriver/Task3/BaseTest.java:
```diff
@@ -0,0 +1,36 @@
+        driver = new DriverFactory().createInstance(browser);
+    }
+//
+//    @AfterEach
```
don't leave commented code in the classes

vaddost commented on src/test/java/com/epam/training/student_anton_lapushenko/WebDriver/Task3/GoogleCloudCalculatorTest.java:
```diff
@@ -0,0 +1,42 @@
+
+
+    @Test
+    public void testGoogleCloudCalculator(){
```
this test don't have any verifications and it is not stable. Each test should contain at least one verification

vaddost commented on src/test/java/com/epam/training/student_anton_lapushenko/WebDriver/Task3/GoogleCloudCalculatorTest.java:
```diff
@@ -0,0 +1,42 @@
+    }
+
+    @Test
+    public void testGoogleCalculator(){
```
the same, test doesn't contain any verifications

mccreed13 committed:
Fixed after comments.
Added Factory, EstimateSteps, Enums, SummaryPage and implicit wait. Avoided absolut paths.
